cooked-validators/src/Cooked/Tweak/Common.hs

```diff
@@ -92,26 +91,31 @@ setTweak optic newValue = getTxSkel >>= putTxSkel . set optic newValue
 overTweak :: (MonadTweak m, Is k A_Setter) => Optic' k is TxSkel a -> (a -> a) -> m ()
 overTweak optic change = getTxSkel >>= putTxSkel . over optic change
 
+-- | The tweak that modifies values satisfying a predicate in the 'TxSkel'.
+-- Returns a list of the foci that were modified prior to the modification.
+overPredTweak :: (MonadTweak m, Is k A_Traversal) => Optic' k is TxSkel a -> (a -> a) -> (a -> Bool) -> m [a]
```

Comment:
This change is motivated by the use cases we had in optim where we would very often want to make modifications based on wether some element satisfies some predicate rather than encompassing this distinction in a `Maybe`. I think it's pretty clear that the predicate filters the foci though, I don't see how it could impact the outcomes of a transformation. I will try and make it even more explicit in a comment.

Reply:
I understand and conquer that this interpretation is possible as well.
This change is not based on anything justifiable on a conceptual level from me. It's purely practical. In my last audit, 50% of the tweaks I used were of the form where I would want to apply a given transformation if the foci satisfies a certain condition, which is why I would like to have this in the API. If this specific implementation seems vague to handle these use cases, maybe we can have optics with builtin conditions instead?

Reply:
Great, because this solution makes me even happier than my first attempt as it is conceptually more satisfying while retaining the same capabilities !
Shall I add it to the PR ?